data/service/api/v1/alerts.go

```diff
@@ -119,23 +120,21 @@ func UpsertAlert(dCtx service.Context) {
 		return
 	}
 
-	followedUserID := r.PathParam("followedUserID")
-	userID := userIDWithServiceFallback(authDetails, r.PathParam("userID"))
 	pc := dCtx.PermissionClient()
-	if err := checkUserAuthorization(ctx, pc, userID, followedUserID); err != nil {
+	if err := checkUserAuthorization(ctx, pc, userIDWithServiceFallback(authDetails, path.UserID), path.FollowedUserID); err != nil {
 		dCtx.RespondWithError(platform.ErrorUnauthorized())
 		return
 	}
 
-	cfg := &alerts.Config{UserID: userID, FollowedUserID: followedUserID, Alerts: *a}
+	cfg := &alerts.Config{UserID: path.UserID, FollowedUserID: path.FollowedUserID, Alerts: *a}
 	if err := repo.Upsert(ctx, cfg); err != nil {
 		dCtx.RespondWithError(platform.ErrorInternalServerFailure())
 		lgr.WithError(err).Error("upserting alerts config")
 		return
 	}
 }
 
-var ErrUnauthorized = fmt.Errorf("unauthorized")
+var ErrUnauthorized = platformerrors.New("unauthorized")
```

Comment:
Please just create the error where it is needed. Using a "global" variable breaks the cause/caller "stack" for the error itself. (That is, it shows this location and not where the error actually originated.)

Reply:
Here I thought I was doing right by not using fmt... 

How do you recommend I alter tests to check for that error? I'll try to see how it's done in other places, but I've typically used package level errors, as they're what I can actually test for. It's what's done all over the Go stdlib. I understand we have a different errors package though, so continue to have patience with me. :)

Reply:
Its not ideal, but you can use `MatchError` for something like:
```
Expect(err).To(MatchError("unauthorized"))
```

Reply:
Done